django/thunderstore/repository/api/experimental/serializers.py

```diff
@@ -95,3 +98,104 @@ class Meta:
             "has_nsfw_content",
             "categories",
         )
+
+
+class PackageUploadAuthorNameField(serializers.SlugRelatedField):
+    """Package upload's author name metadata field."""
+
+    def __init__(self, *args, **kwargs):
+        kwargs["slug_field"] = "name"
+        super().__init__(*args, **kwargs)
+
+    def get_queryset(self):
+        return UploaderIdentity.objects.filter(
+            members__user=self.context["request"].user,
+        )
+
+
+class PackageUploadCategoriesField(serializers.RelatedField):
+    """Package upload's categories metadata field."""
+
+    def get_queryset(self):
+        return PackageCategory.objects.filter(
```

Comment:
Use exclusion rather than filter when generating a "base" queryset, since chained filters don't often work as you'd expect them to in Django